### PR TITLE
pythonPackages.advantage_air: init at 0.2.1

### PR DESCRIPTION
**Pull request by JamieMagee**
###### Motivation for this change

Support the `advantage_air` component in Home Assistant.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107894` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.advantage_air</li>
    <li>python38Packages.advantage_air</li>
    <li>python39Packages.advantage_air</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107894` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.advantage_air</li>
    <li>python38Packages.advantage_air</li>
    <li>python39Packages.advantage_air</li>
  </ul>
</details>

File: pkgs/development/python-modules/advantage_air/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ aiohttp, buildPythonPackage, fetchPypi, lib, pythonOlder }:
+
+buildPythonPackage rec {
+  pname = "advantage_air";
+  version = "0.2.1";
```
Comment (fabaff) [inline pkgs/development/python-modules/advantage_air/default.nix:5]: The latest release is [0.2.2](https://github.com/Bre77/advantage_air/commit/f7963d16ea782e84595b55663971d0f303586dec).

Comment (JamieMagee) [reply pkgs/development/python-modules/advantage_air/default.nix:5]: Haha, I wasn't expecting the dev to get back to my issue so quickly. See Bre77/advantage_air#1 for context.

Comment (fabaff) [reply pkgs/development/python-modules/advantage_air/default.nix:5]: Yeah, 0.2.2 was shortly released after this PR was made.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -171,6 +171,8 @@ in {
 
   adguardhome= callPackage ../development/python-modules/adguardhome { };
 
+  advantage_air= callPackage ../development/python-modules/advantage_air { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:174]: Please normalize the attribute name. According to the [python contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines), package names within nix should follow [PEP 0503](https://www.python.org/dev/peps/pep-0503/#normalized-names)
```suggestion
  advantage-air= callPackage ../development/python-modules/advantage-air { };
```

the pname can remain as-is though.

File: pkgs/servers/home-assistant/component-packages.nix [hunk 3]
```diff
@@ -11,7 +11,7 @@
     "actiontec" = ps: with ps; [ ];
     "adguard" = ps: with ps; [ adguardhome ];
     "ads" = ps: with ps; [ pyads ];
-    "advantage_air" = ps: with ps; [ ]; # missing inputs: advantage_air
+    "advantage_air" = ps: with ps; [ advantage_air ];
```
Comment (SuperSandro2000) [inline pkgs/servers/home-assistant/component-packages.nix:14]:
```suggestion
    "advantage_air" = ps: with ps; [ advantage-air ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/servers/home-assistant/component-packages.nix (hunk 3 above)
Comment (mweinelt) [reply pkgs/servers/home-assistant/component-packages.nix:14]: This file is autogenerated, so please execute `parse-requirements.py` to update it.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107894` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.advantage-air</li>
<li>python38Packages.advantage-air</li>
<li>python39Packages.advantage-air</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107894` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.advantage-air</li>
<li>python38Packages.advantage-air</li>
<li>python39Packages.advantage-air</li>
</ul>
</details>

File: pkgs/servers/home-assistant/component-packages.nix (hunk 3 above)
Comment (mweinelt) [reply pkgs/servers/home-assistant/component-packages.nix:14]: Not sure why this was marked resolved, it was still in need of the change.

Resolved in https://github.com/NixOS/nixpkgs/pull/108982.
